exist-core/src/main/java/org/exist/dom/persistent/LockToken.java

```diff
@@ -146,29 +123,136 @@ public static String generateUUID() {
     }
 
     public void write(final VariableByteOutputStream ostream) throws IOException {
-        ostream.writeByte(type);
-        ostream.writeByte(depth);
-        ostream.writeByte(scope);
+        // TODO(AR) these 3 bytes could be encoded into 1
+        ostream.writeByte(type.getValue());
+        ostream.writeByte(depth.getValue());
+        ostream.writeByte(scope.getValue());
+
         ostream.writeUTF(owner != null ? owner : "");
         ostream.writeLong(timeout);
         ostream.writeUTF(token != null ? token : "");
-        ostream.writeByte(resourceType);
+        ostream.writeByte(resourceType.getValue());
     }
 
-    public void read(final VariableByteInput istream) throws IOException {
-        type = istream.readByte();
-        depth = istream.readByte();
-        scope = istream.readByte();
-        owner = istream.readUTF();
+    public static LockToken read(final VariableByteInput istream) throws IOException {
```

Comment:
Ideally... this meta data would be stored in a non-shared dbx file or so? I mean, per document/collection quite a lot of bytes are preserved PLUS a string representation of the lock-owner.